### PR TITLE
Add logging to the API caller

### PR DESCRIPTION
**Pull request by ccostino**
This changeset adds logging to the API caller so we have a bit more visibility into things when an error occurs.  The logging takes place in the underlying `_call_api` method that serves as the backbone for all API requests made by the CMS.

There are several formatting improvements included in this changeset to help improve readability as well.

h/t to @LindsayYoung, who originally did this work!

### CODE REVIEW

File: fec/data/api_caller.py [hunk 1]
```diff
@@ -1,21 +1,26 @@
-from django.conf import settings
```
Comment (xtine) [inline fec/data/api_caller.py]: Nice, thanks for rearranging imports.

File: fec/data/api_caller.py [hunk 2]
```diff
@@ -25,13 +30,23 @@ def _call_api(*path_parts, **filters):
...
+    path = os.path.join(
+        settings.FEC_API_VERSION,
+        *[x.strip('/') for x in path_parts]
+    )
     url = parse.urljoin(settings.FEC_API_URL, path)
-
     results = session.get(url, params=filters)
 
-    return results.json() if results.ok else {}
+    if results.ok:
```
Comment (xtine) [inline fec/data/api_caller.py]: Much easier to read, and now logging! 👍

Comment (ccostino) [reply fec/data/api_caller.py]: I was definitely getting Ruby vibes with that line as it was... 😆

**Comment by lbeaufort** [comment]
@ccostino this looks so great, good thought. I plan to test this locally tomorrow, but I don't anticipate any problems.

**Comment by ccostino** [comment]
Yes it's ready and yes `develop` is okay, thank you @lbeaufort! :-)

**Pull request merged**